### ISSUE TITLE
Adding Authorization Header as interceptor

### ISSUE THREAD

**Issue by Pirlo220** [andreareginato/oauth-ng#68]
Hi, I'm reviewing your API and cannot find a way to include for every http request an authorization header with the Bearer and access_token.

I've modified the ExpiredInterceptor ('oauth.interceptor') to add this: 
if (token && expired(token)){
      $rootScope.$broadcast('oauth:expired', token);
} else if (token !== undefined){
        config.headers.Authorization = 'Bearer ' + token.access_token;     
};

Do you think this is affortable for coming versions??

**Comment by tonqa** [comment]
+1

works perfectly, thanks Pirlo.

**Comment by flea19** [comment]
Will this be included in a future release? Or is the default that a custom interceptor that includes teh bearer token must be provided.

**Comment by m00s** [comment]
Hi guys and thanks @Pirlo220 for the suggestion. It looks good to me too.
It would be very appreciated if someone of you could provide a PR for this

**Comment by bsideup** [comment]
You this instead:

```javascript
angular.module('app').run(function($rootScope, $http) {
    function setToken(token) {
      if(!token) {
        delete $http.defaults.headers.common.Authorization;
      } else {
        $http.defaults.headers.common.Authorization = 'Bearer ' + token.access_token;
      }
      
      console.log("OAuth token set to: " + token)
    }
    
    $rootScope.$on('oauth:authorized', function(event, token) { setToken(token) });
    $rootScope.$on('oauth:loggedOut', function(event) { setToken(null) })
    $rootScope.$on('oauth:expired', function(event) { setToken(null) })
  })
```

**Comment by m00s** [comment]
closed by #98
